sql/src/main/java/io/crate/analyze/CreateAnalyzerStatementAnalyzer.java

```diff
@@ -21,240 +21,156 @@
-                // type mandatory
-                String evaluatedType = extractType(properties, context.analysis.parameterContext());
-                if (!context.statement.analyzerService().hasBuiltInTokenizer(evaluatedType)) {
-                    // only builtin tokenizers can be extended, for now
-                    throw new IllegalArgumentException(String.format(Locale.ENGLISH,
-                        "Non-existing built-in tokenizer type '%s'", evaluatedType));
-                }
-            } else {
-                throw new IllegalArgumentException(String.format(Locale.ENGLISH, "tokenizer name '%s' is reserved", name));
-            }
-            // build
-            // transform name as tokenizer is not publicly available
-            name = String.format(Locale.ENGLISH, "%s_%s", context.statement.ident(), name);
-
-            Settings.Builder builder = Settings.builder();
-            for (Map.Entry<String, Expression> tokenizerProperty : properties.properties().entrySet()) {
-                GenericPropertiesConverter.genericPropertyToSetting(builder,
-                    TOKENIZER.buildSettingChildName(name, tokenizerProperty.getKey()),
-                    tokenizerProperty.getValue(),
-                    context.analysis.parameterContext().parameters());
-            }
-            context.statement.tokenDefinition(name, builder.build());
+            extendedAnalyzerName = null;
         }
 
-        return null;
-    }
-
-    @Override
-    public CreateAnalyzerAnalyzedStatement visitGenericProperty(GenericProperty<?> property, Context context) {
-        GenericProperty<Expression> property1 = (GenericProperty<Expression>) property;
-        GenericPropertiesConverter.genericPropertyToSetting(
-            context.statement.genericAnalyzerSettingsBuilder(),
-            ANALYZER.buildSettingChildName(context.statement.ident(), property1.key()),
-            property1.value(),
-            context.analysis.parameterContext().parameters()
+        var analyzedCreateAnalyzer = new AnalyzedCreateAnalyzer(analyzerIdent, extendedAnalyzerName);
+        var context = new Context(
+            analyzedCreateAnalyzer,
+            functions,
+            transactionContext,
+            convertParamFunction
         );
-        return null;
-    }
 
-    @Override
-    public CreateAnalyzerAnalyzedStatement visitTokenFilters(TokenFilters tokenFilters, Context context) {
-        for (NamedProperties tokenFilterNode : tokenFilters.tokenFilters()) {
+        for (AnalyzerElement<Expression> element : createAnalyzer.elements()) {
+            AnalyzerElementsAnalysisVisitor.analyze(element, context);
+        }
+        return analyzedCreateAnalyzer;
+    }
 
-            String name = tokenFilterNode.ident();
-            GenericProperties<Expression> properties = tokenFilterNode.properties();
+    private static class AnalyzerElementsAnalysisVisitor
```

Comment:
I've introduced the `AnalyzerElementsAnalysisVisitor` only for the analyzer elements, before the `CreateAnalyzerStatementAnalyzer` was extending the visitor methods including the `visitCreateAnalyzer` which seems to be not that needed and a bit confusing.